1.5/Source/PawnStorages/PawnStoragesModule-DefensivePositions/CompPawnStorage_Patch.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using DefensivePositions;
+using HarmonyLib;
+using RimWorld;
+using UnityEngine;
+using Verse;
+
+namespace PawnStorages.DefensivePositions;
+
+[HarmonyPatch(typeof(CompPawnStorage))]
+public static class CompPawnStorage_Patch
+{
+    [HarmonyPatch(nameof(CompPawnStorage.CompGetGizmosExtra))]
+    [HarmonyPostfix]
+    public static void CompGetGizmosExtra_Patch(CompPawnStorage __instance, ref IEnumerable<Gizmo> __result)
+    {
+        Command_Action action = new Command_Action();
+        action.defaultLabel = "PS_ReleaseAllToDefensivePositions".Translate();
+        action.defaultDesc = "PS_ReleaseAllToDefensivePositions_Tooltop".Translate();
+        action.icon = ContentFinder<Texture2D>.Get("UIPositionLargeActive", true);
+
+        action.action = delegate
+        {
+
+            for (int num = __instance.innerContainer.Count - 1; num >= 0; num--)
+            {
+                Pawn pawn = (Pawn) __instance.GetAt(num);
+                __instance.innerContainer.Remove(pawn);
+                GenSpawn.Spawn(pawn, __instance.parent.Position, __instance.parent.Map);
```

Comment:
I think the solution is to spawn them on the interaction cell instead of the wall mounted item cell
if it doesn't have an interaction cell there's a place-near instead which will find an accessible spot to place them

Reply:
I put interaction spot code into `Utility.ReleasePawn` and changed this code to use that instead.